pkg/utils/transform/transform_test.go

```diff
@@ -0,0 +1,53 @@
+package transform
+
+import (
+	"github.com/tektoncd/operator/pkg/flag"
+	"testing"
+
+	mf "github.com/jcrossley3/manifestival"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func TestTransformManifest_WithAnnotation(t *testing.T) {
+	resourceWithAnnotation := "../../../test/testdata/resource-with-annotation.yaml"
+	manifest, err := mf.NewManifest(resourceWithAnnotation, true, nil)
+	if err != nil {
+		t.Errorf("NewManifest() = %v, wanted no error", err)
+	}
+
+	tfs := []mf.Transformer{
+		InjectNamespaceConditional(flag.AnnotationPreserveNS, "target"),
+	}
+
+	if err := manifest.Transform(tfs...); err != nil {
+		t.Error("wanted no error :", err)
+	}
+
+	assertNamespace(manifest.Resources[0], "openshift", t)
+}
+
+func TestTransformManifest_WithoutAnnotation(t *testing.T) {
+	resourceWithoutAnnotation := "../../../test/testdata/resource-without-annotation.yaml"
+	manifest, err := mf.NewManifest(resourceWithoutAnnotation, true, nil)
+	if err != nil {
+		t.Errorf("NewManifest() = %v, wanted no error", err)
+	}
+
+	tfs := []mf.Transformer{
+		InjectNamespaceConditional(flag.AnnotationPreserveNS, "target"),
+	}
+
+	if err := manifest.Transform(tfs...); err != nil {
+		t.Error("wanted no error :", err)
+	}
+
+	assertNamespace(manifest.Resources[0], "target", t)
+}
+
+func assertNamespace(u unstructured.Unstructured, expected string, t *testing.T) {
+	v, _, _ := unstructured.NestedMap(u.Object, "metadata")
```

Comment:
add a `t.Helper()` as the first line in this assert function
for better test logs